portal.ejb/src/main/java/org/vamdc/portal/session/queryBuilder/formsTree/ReactionCollisionForm.java

```diff
@@ -0,0 +1,163 @@
+package org.vamdc.portal.session.queryBuilder.formsTree;
+
+import org.vamdc.portal.session.queryBuilder.QueryTreeInterface;
+import org.vamdc.portal.session.queryBuilder.fields.AbstractField;
+import org.vamdc.portal.session.queryBuilder.forms.AbstractForm;
+import org.vamdc.portal.session.queryBuilder.forms.FormForFields;
+import org.vamdc.portal.session.queryBuilder.forms.Order;
+
+public abstract class ReactionCollisionForm extends AbstractForm implements FormForFields, TreeFormInterface{
+
+
+	/**
+	 * 
+	 */
+	private static final long serialVersionUID = -6587803873839744100L;
+	protected Integer typeAAtomCount;
+	protected Integer typeAMoleculeCount;
+	protected Integer typeAParticleCount;
+	protected Integer typeBAtomCount;
+	protected Integer typeBMoleculeCount;
+	protected Integer typeBParticleCount;	
+	protected Integer maxElementCount = 5;
+	private QueryTreeInterface tree;
+	private Boolean queryable = true;
+		
+	//private Boolean hasHiddenForm = false;
+	
+	private String typeAName;
+	private String typeBName;
+	
+	public ReactionCollisionForm(QueryTreeInterface tree, String typeA, String typeB){
+		this.tree = tree;
+		this.typeAName = typeA;
+		this.typeBName = typeB;
+		
+	}
+	
+	/*private void addHiddenForms(){
+		
+		if( false == this.hasHiddenForm){		
+			AtomsTreeForm f = new AtomsTreeForm(tree, typeAName);
+			f.setVisible(false);
+			tree.addForm(f);
+			 this.hasHiddenForm = true;		 
+		}
+	}*/
+	
+
+	@Override
+	public void validate() {
+		//this.addHiddenForms();
+		
+		for(int i=0;i<typeAAtomCount;i++){
```

Comment:
I'm wondering if in Java there is a fancier way to execute code N times, then writing a for loop? Probably not :(